api/src/chat/controllers/block.controller.ts

```diff
@@ -64,6 +67,38 @@ export class BlockController extends BaseController<
     super(blockService);
   }
 
+  /**
+   * Text search for blocks using MongoDB text index.
+   *
+   * Example: GET /block/search?q=UserSearchPhrase&limit=50
+   */
+  @Get('search')
+  async search(
+    @Query('q') q?: string,
+    @Query('limit') limitQuery?: string,
+    @Query('category') category?: string,
```

Comment:
Please check other controllers, I believe there is a pipe / filter that allows to validate that category is indeed an object ID.